File: frontend/lib/tmdb.ts
// use axios to fetch data from tmdb
//curl --location 'https://api.themoviedb.org/3/search/movie?api_key=202cf556ed2f3cd393839ea81335542d&query=Ratatouille&language=en-US'

import axios from "axios";

const API_URL = "https://api.themoviedb.org/3";
const API_KEY = "202cf556ed2f3cd393839ea81335542d";

export async function searchMovies(query: string) {
  const response = await axios.get(`${API_URL}/search/movie`, {
    params: {
      api_key: API_KEY,
      query,
      language: "en-US",
      include_adult: false,
    },
  });
  return response.data;
}

// curl --location 'https://api.themoviedb.org/3/movie/=/reviews?api_key=202cf556ed2f3cd393839ea81335542d&language=en-US'

export async function getMovieReviews(movieId: number) {
  const response = await axios.get(`${API_URL}/movie/${movieId}/reviews`, {
    params: {
      api_key: API_KEY,
      language: "en-US",
    },
  });
  return response.data;
}

File: backend/app/services/tmdb.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

api_key = os.getenv("TMDB_API_KEY")

def search_movie(query: str):
    
    url = "https://api.themoviedb.org/3/search/movie"
    params = {
        "api_key": api_key,
        "query": query,
        "language": "en-US",
        "page": 1,
        "include_adult": False
    }
    
    response = requests.get(url, params=params)
    if response.status_code != 200:
        raise Exception(f"TMDB API request failed with status code {response.status_code}")
    
    return response.json()

def get_movie_details(movie_id: int):
    
    url = f"https://api.themoviedb.org/3/movie/{movie_id}"
    params = {
        "api_key": api_key,
        "language": "en-US"
    }
    response = requests.get(url, params=params)
    if response.status_code != 200:
        raise Exception(f"TMDB API request failed with status code {response.status_code}")
    return response.json()

#call function from cmd line
if __name__ == "__main__":
    query = "Ratatouille"
    results = search_movie(query)
    print(results)